### PR TITLE
Updated OAuth2 init check and added unit test

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Updated OAuth2 init check and added unit test

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/17749

### CODE REVIEW

File: components/eventing-controller/controllers/backend/reconciler.go [hunk 1]
```diff
@@ -1001,6 +1001,12 @@ func (r *Reconciler) getMutatingAndValidatingWebHookConfig(ctx context.Context)
 }
 
 func (r *Reconciler) isOauth2CredentialsInitialized() bool {
+	if featureflags.IsEventingWebhookAuthEnabled() {
```
Comment (marcobebway) [inline components/eventing-controller/controllers/backend/reconciler.go:1004]: 👍

File: components/eventing-controller/controllers/backend/reconciler.go [hunk 2]
```diff
@@ -1001,6 +1001,12 @@ func (r *Reconciler) getMutatingAndValidatingWebHookConfig(ctx context.Context)
...
+		return len(r.credentials.clientID) > 0 &&
+			len(r.credentials.clientSecret) > 0 &&
+			len(r.credentials.tokenURL) > 0 &&
+			len(r.credentials.certsURL) > 0
+	}
```
Comment (marcobebway) [inline components/eventing-controller/controllers/backend/reconciler.go:1009]: 👍

File: components/eventing-controller/controllers/backend/reconciler_internal_unit_test.go [hunk 3]
```diff
@@ -549,6 +549,66 @@ func Test_getOAuth2ClientCredentials(t *testing.T) {
...
+			givenFlagEnabled: true,
+			givenCredentials: &oauth2Credentials{
+				clientID:     []byte("test"),
+				clientSecret: []byte("test"),
+				certsURL:     []byte("http://kyma-project.io"),
+				tokenURL:     []byte("http://kyma-project.io"),
+			},
+			wantResult: true,
+		},
+	}
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply components/eventing-controller/controllers/backend/reconciler_internal_unit_test.go:639]: added

**Review by marcobebway — APPROVED** [review]

**Pull request merged**